### ISSUE TITLE
Error when generating from sphinx html doc

### ISSUE THREAD

**Issue by hbprotoss** [hynek/doc2dash#37]
I have generated html api doc from `sphinx-apidoc`, but when I use doc2dash to convert to docset, error occurred:

Traceback (most recent call last):
  File "/usr/local/bin/doc2dash", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 572, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 552, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 744, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 388, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/doc2dash/__main__.py", line 115, in main
    docset = prepare_docset(source, dest, name, index_page)
  File "/usr/local/lib/python2.7/site-packages/doc2dash/__main__.py", line 248, in prepare_docset
    shutil.copytree(source, docs)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 208, in copytree
    raise Error, errors
shutil.Error: [('html/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Info.plist', u'doc/scrapy.docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Info.plist', "[Errno 63] File name too long: u'doc/scrapy.docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Info.plist'"), ('html/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources', u'doc/scrapy.docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources', "[Errno 63] File name too long: 'doc/scrapy.docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources/Documents/..docset/Contents/Resources'")]

**Comment by hynek** [comment]
looks like some circular symlink?

**Comment by henriquebastos** [comment]
`doc2dash` rely on the `SOURCE` command line argument to create the `SOURCE.docset` directory.

When I was **inside** the documentation directory, I used `.` as the `SOURCE`. So `doc2dash` created the directory `..docset` inside the documentation directory.

When I tried it a second time defining the `-n mylib` flag. The bogus `..docset` inside the documentation kept `copytree` failing.

This is issue here is not a bug, but maybe `doc2dash` could use `os.path.realpath` inside the `setup_paths` function to make sure this won't happen.
